tripal/tripal
# Add tripal web services http tests

Issue #567

## Type(s) of Change(s)


## Description



Adds unit tests to verify that http calls to web services behave as expected.

## Testing?





1. Be sure to run `composer update` to get the latest version of TripalTestSuite
2. Run `./vendor/bin/phpunit tests/tripal_ws` (there are no side effects to running tests since they are wrapped in a transaction)
3. Verify all assertions are passing (green)

## Screenshots (if appropriate):

## Additional Notes (if any):



I encountered an issue when testing pulling a newly published feature. When sending a GET request to `/web-services/content/v0.1/mRNA/$entity_id`, where `$entity_id` is a newly created mRNA entity, I get an error indicating that the entity is not available. However, using `tripal_load_entity()` finds and loads the entity perfectly fine. Any idea why this test doesn't pass? (see method `testGettingAFeatureResource()` in this PR).

Thanks!
